processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java

```diff
@@ -98,11 +102,25 @@ public DimFilter optimize()
   {
     if (this.getExtractionFn() instanceof LookupExtractionFn
```

Comment:
why not `return new SelectorDimFilter(dimension, value, extractionFn).optimize();`?